db/revision_cache_interface.go

```diff
@@ -32,6 +32,12 @@ type RevisionCache interface {
 	// Put will store the given docRev in the cache
 	Put(docRev DocumentRevision)
 
+	// Update will remove existing value and re-create new one
+	Upsert(docRev DocumentRevision)
+
+	// Invalidate
```

Comment:
Can you provide a description for the use-case or behaviour of Invalidate.

It's not clear what it's used for without knowing what was said in the discussion to implement this, which will be forgotten sooner or later.

Reply:
Added comments how do they look?